runtime/android/core_shell/src/org/xwalk/core/xwview/shell/XWalkViewShellActivity.java

```diff
@@ -359,4 +324,132 @@ public void onXWalkViewCreated(XWalkView view) {
     }
+
+    @Override
+    public void onXWalkCoreReady() {
+        setContentView(R.layout.testshell_activity);
+
+        mActionBar = getActionBar();
+        mActionBar.setNavigationMode(ActionBar.NAVIGATION_MODE_TABS);
+        mSectionsPagerAdapter = new SectionsPagerAdapter(this, getSupportFragmentManager(), mActionBar);
+
+        mViewPager = (ViewPager) findViewById(R.id.pager);
+        mViewPager.setAdapter(mSectionsPagerAdapter);
+        mViewPager.setOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
+            @Override
+            public void onPageSelected(int position) {
+                mActionBar.setSelectedNavigationItem(position);
+            }
+        });
+
+        mProgressMap = new HashMap<XWalkView, Integer>();
+        // Add two tabs.
+        mActionBar.addTab(
+                mActionBar.newTab()
+                        .setText(mSectionsPagerAdapter.getPageTitle(0))
+                        .setTabListener(this));
+        mActionBar.addTab(
+                mActionBar.newTab()
+                        .setText(mSectionsPagerAdapter.getPageTitle(1))
+                        .setTabListener(this));
+
+        mToolbar = (LinearLayout) findViewById(R.id.toolbar);
+        mProgressDrawable = (ClipDrawable) findViewById(R.id.toolbar).getBackground();
+
+        IntentFilter intentFilter = new IntentFilter(ACTION_LAUNCH_URL);
+        mReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                Bundle bundle = intent.getExtras();
+                if (bundle == null)
+                    return;
+
+                if (bundle.containsKey("url")) {
+                    String extra = bundle.getString("url");
+                    if (mActiveView != null)
+                        mActiveView.load(sanitizeUrl(extra), null);
+                }
+            }
+        };
+        registerReceiver(mReceiver, intentFilter);
+    }
+
+    @Override
+    public void onXWalkCoreStartupError(Throwable e, XWalkCoreStatus status) {
+        if (status == XWalkCoreStatus.COMPRESSED) {
+            Dialog dialog = new StartupDecompressDialog(this);
+            dialog.show();
+        }
+    }
+
+    @Override
+    public void onXWalkCoreRuntimeError(Throwable e, XWalkCoreStatus status) {
+    }
+
+    @Override
+    public void reserveReflectObject(Object object) {
+    }
+
+    @Override
+    public void reserveReflectMethod(ReflectMethod method) {
+    }
+
+    private static class StartupDecompressDialog extends AlertDialog {
+        XWalkViewShellActivity mActivity;
+
+        StartupDecompressDialog(XWalkViewShellActivity activity) {
+            super(activity);
+            mActivity = activity;
+
+            OnClickListener positiveListener = new OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialog, int id) {
+                    mActivity.finish();
```

Comment:
The asyncTask is still running in background until completion even if the activity has been finished. It's necessary to call asyncTask.cancel(). And do not forget to check isCancelled() for every loop in doInBackground()